asab/utils.py

```diff
@@ -28,3 +31,27 @@ def convert_to_seconds(value: str) -> float:
 		raise ValueError("'{}' is not a valid time specification: {}.".format(value, e))
 
 	return value
+
+
+def validate_url(input_url: str, scheme):
+	# Remove leading and trailing whitespaces before parsing
+	url = urlparse(input_url.strip())
+
+	if url.path.endswith("/"):
+		url = url._replace(path=url.path[:-1])
+
+	if scheme is None:  # Scheme doesn't get checked
+		return url.geturl()
+	elif isinstance(scheme, tuple):  # Supports tuple
+		if url.scheme in scheme:
+			return url.geturl()
+	elif scheme == url.scheme:
+		return url.geturl()
+	else:
+		if url.scheme:
+			raise ValueError("{} has an invalid scheme: {}".format(url.geturl(), url.scheme))
```

Comment:
Using " '{}' " is more friendly when string is empty.

Reply:
Fixed